elytron/ELY-1712-enhanced-audit-logging.adoc

```diff
@@ -0,0 +1,73 @@
+= [WFCORE-4394] Enhanced Audit Logging - RFC support and reliability/speed customization
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, Audit Logging in Elytron does not have the same capabilities as Legacy Security. This task is to enhance Elytron Audit Logging to have the same capabilities as Legacy Security Audit Logging.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4394[WFCORE-4394]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-699[EAP7-699]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jkasik@redhat.com[Jan Kašík]
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Support for https://tools.ietf.org/html/rfc5424[RFC 5424] should be enhanced, according to the following parameter,
+which should be added to ```syslog-audit-log``` in the Elytron subsystem
+** A ```syslog-format``` attribute should be added which supports a value of ```RFC5424```
+** This attribute will also have ```RFC5424``` as the default value
+** A Rfc5424SyslogEvent should be added as an audit event, which will trigger when the Elytron Subsystem attempts to log to a syslog server with the format ```RFC5424```
+
+* Support for https://tools.ietf.org/html/rfc3164[RFC 3164] should be added to ```syslog-audit-log``` in the Elytron subsystem
+** The ```syslog-format``` attribute described in the RFC 5424 section above should also support a value of ```RFC3164```
+** A Rfc3164SyslogEvent should be added as an audit event, which will trigger when the Elytron Subsystem attempts to log to a syslog server with the format ```RFC3164```
+
+* The ability to configure for less reliability for increased speed should be added
+** This should be done by configuring the number of reconnect attempts to a syslog-server
+** This should be available as an attribute ```reconnect-attempts``` under ```syslog-audit-log``` in the Elytron subsystem
+** This would be an optional parameter and not have a default value
```

Comment:
Just a very last thing. Wouldn't it be better if this has a default value of current count of reconnect attempts and making this a required attribute?

Reply:
Updated this section to still be optional but have a default value of 0

Reply:
Thank you. Sorry, I meant "required" in sense, that it will have default value presenting current amount of reconnect attempts. Users won't have to set anything if they don't want to. If it will have default value of `0`, does it mean, that there will be no reconnect attempts made, or there will be infinite amount of them?

Reply:
By 0 I mean there would be no reconnect attempts

Reply:
So how many reconnect attempts is done now?

Reply:
Currently it indefinitely attempts to reconnect

Reply:
In that case, wouldn't be the value of `-1` best? If user wishes to change the behavior, he can set any positive number (thus swapping reliability for speed).

Reply:
Ya that would be better, I'll change it to -1

Reply:
Updated the description